rmw_zenoh_cpp/src/zenohd/main.cpp

```diff
@@ -217,6 +218,8 @@ int main(int argc, char ** argv)
       c = keyreader.readOne();
     } catch (const std::runtime_error &) {
       perror("read():");
+
+      z_close(z_move(s));
```

Comment:
to prevent duplicated `z_close(z_move(s));` calls (here and below in line 233), I wonder if it's better to wrap this statement within a `rcpputils::scope_exit` block like https://github.com/ros2/rmw_zenoh/blob/fcfed30ca7ba629a137bc48fab0502c71e515839/rmw_zenoh_cpp/src/rmw_init.cpp#L203-L206

Reply:
Added scope_exit functionality in this commit ( https://github.com/ros2/rmw_zenoh/pull/252/commits/0566dc2a4736931e8c61c3bd370cee78ee07716b )